src/SquawkLinter.php

```diff
@@ -0,0 +1,157 @@
+    $options = array(
+      'squawk.config' => array(
+        'type' => 'string',
+        'help' => pht('Configuration file to use'),
+      ),
+    );
+    return $options + parent::getLinterConfigurationOptions();
+  }
+
+  public function setLinterConfigurationValue($key, $value) {
+    switch ($key) {
+      case 'squawk.config':
+        $this->config = $value;
+        return;
+    }
+    return parent::setLinterConfigurationValue($key, $value);
+  }
+
+  public function getNodeBinary() {
+    return 'squawk';
+  }
+
+  public function getNpmPackageName() {
+    return 'squawk-cli';
+  }
+
+  protected function getMandatoryFlags() {
+    return array('-c', $this->config);
+  }
+
+  protected function getDefaultFlags() {
+    return array('--reporter', 'Json');
+  }
+
+  public function shouldExpectCommandErrors() {
+    return true;
+  }
+
+  public function getLintSeverityMap() {
+    return array(
+      'Error' => ArcanistLintSeverity::SEVERITY_ERROR,
+      'Warning' => ArcanistLintSeverity::SEVERITY_WARNING
+    );
+  }
+
+  protected function parseLinterOutput($path, $err, $stdout, $stderr) {
+    /*
+      [
+      	{
+      		"file": "path/to.sql",
+      		"line": 0,
+      		"column": 0,
+      		"level": "Error",
+      		"messages": [
+      			{
+      				"Note": "Postgres failed to parse query: syntax error at or near \"some_random_stuffs\""
+      			},
+      			{
+      				"Help": "Modify your Postgres statement to use valid syntax."
+      			}
+      		],
+      		"rule_name": "invalid-statement"
+      	},
+      	{
+      		"file": "path/to_second.sql",
+      		"line": 1,
+      		"column": 0,
+      		"level": "Warning",
+      		"messages": [
+      			{
+      				"Note": "Dropping a column may break existing clients."
+      			}
+      		],
+      		"rule_name": "ban-drop-column"
+      	}
+      ]
+    */
+    $json = json_decode($stdout, true);
+
+    $messages = array();
+    foreach ($json as $item) {
+      $level = $item['level'];
+      $rulename = $item['rule_name'];
+      $description = "$level ($rulename): ";
+
+      foreach($item['messages'] as $message) {
+        $value = '';
+
+        if (array_key_exists('Note', $message)) {
+          $value = $message['Note'];
+        }
+
+        if (array_key_exists('Help', $message)) {
+          $value = $message['Help'];
+        }
+
+        $description.="\n$value";
+      }
+
+      $line = $item['line'];
+      $column = $item['column'];
+
+      // $line=0 means the file has broken syntax
+      // otherwise sum the $line and $ column to get the first row of error
+      if ($line == '0') {
+        $line = 1;
+      } else if ($column != '0') {
+        $line = $line + $column - 1;
+      }
+
+      $messages[] = id(new ArcanistLintMessage())
+        ->setName($this->getLinterName())
+        ->setCode($rulename)
+        ->setPath(idx($item, 'source', $path))
+        ->setLine($line)
+        ->setChar(1)
```

Comment:
I would have expecting `$column` to be used here to set the "character" offset, but it looks like it's being used to adjust the `$line` above?

Reply:
the `$column` here is showing how many rows are being warned by the linter
for example you have drop column like this:
```
ALTER TABLE

```

Reply:
the `$column` here is showing how many rows are being warned by the linter
for example you have drop column like this:
```sql
   1 ALTER TABLE
   2   table_name
   3 DROP COLUMN
   4   column_name;
   5 
   6 -- test some column with spaces
   7               ALTER TABLE table_name DROP COLUMN column_name;
   8 
   9 
  10 -- test column with comment
  11 /* bla3 */ ALTER TABLE table_name DROP column column_name;
```

the first one will have line 1 column 0
the second one will have line 5 column 2 (so I added line + column - 1)
the third one will have line 8 column 3 (8+3-1 = 10)
to show which line is the starting one (includes comment)

and the char offset is not shown in the linter, so it's put as one